akka-javasdk/src/main/scala/akka/javasdk/impl/keyvalueentity/KeyValueEntitiesImpl.scala

```diff
@@ -168,8 +168,10 @@ private[impl] final class KeyValueEntitiesImpl(
             val cmdPayloadPbAny = command.payload.getOrElse(
               // FIXME smuggling 0 arity method called from component client through here
               ScalaPbAny.defaultInstance.withTypeUrl(AnySupport.JsonTypeUrlPrefix).withValue(ByteString.empty()))
-            val cmdBytesPayload = AnySupport.toSpiBytesPayload(cmdPayloadPbAny)
-            val cmd = service.serializer.fromBytes(cmdBytesPayload)
+            // FIXME shall we deserialize here or in the router? the router needs the contentType as well.
+//            val cmdBytesPayload = AnySupport.toSpiBytesPayload(cmdPayloadPbAny)
+//            val cmd = service.serializer.fromBytes(cmdBytesPayload)
+            val cmd = cmdPayloadPbAny
```

Comment:
The original was:
```
val cmd = service.messageCodec.decodeMessage
```
I can't tell how many times I have run into that trap. That was not decoding at all for the normal JsonMessageCodec, so it was still PbAny passed down to the router.